httpsync/message.go

```diff
@@ -0,0 +1,96 @@
+package httpsync
+
+import (
+	"encoding/json"
+	"errors"
+	"io"
+
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+)
+
+// HeadMsg represents the head cid from the src.
+type HeadMsg struct {
+	CidStr string
+}
+
+// SignedHeadMsgEnvelope is the signed envelope of a HeadMsg. It includes the
+// public key of the signer so the receiver can verify it and convert it to a
+// peer id. Note, the receiver is not required to use the provided public key.
+// Also note the signature is over the whole marshalled headmsg instead of just
+// a cidStr. This protects us from forgetting to sign something in the future if
+// we add to HeadMsg.
+type SignedHeadMsgEnvelope struct {
+	HeadMsg []byte
+	Sig     []byte
+	PubKey  []byte
+}
```

Comment:
consider this data struct set up via ipld bind node. & serialized with ipld / dag-json.

i think the wire format change is the cid instead of just being a string would end up dag-json serialized to `{"/":{"cid"}}`, it means that these signed objects can get dumped into the datastore and synced subsequently without special care.

Reply:
Can you elaborate a bit more on this? I don't follow why we'd want to be able to put this into the datastore. The client only needs to open this and read the head message.

Reply:
having one code path for serialization / de-serialization is one less code path to go wrong, but not a strong ask

Reply:
Might be a good exercise. And it could explicitly encode how the bytes are represented (as opposed to the implicit base64 conversion).